pydantic/dataclasses.py

```diff
@@ -24,23 +24,29 @@ def setattr_validate_assignment(self, name, value):
     object.__setattr__(self, name, value)
 
 
-def _process_class(_cls, init, repr, eq, order, unsafe_hash, frozen, validate_assignment):
+def _process_class(_cls, init, repr, eq, order, unsafe_hash, frozen, config):
     post_init_original = getattr(_cls, '__post_init__', None)
     _cls.__post_init__ = post_init
     cls = dataclasses._process_class(_cls, init, repr, eq, order, unsafe_hash, frozen)
 
     fields = {name: (field.type, field.default) for name, field in cls.__dataclass_fields__.items()}
     cls.__post_init_original__ = post_init_original
-    cls.__pydantic_model__ = create_model(cls.__name__, **fields)
+
+    cls.__pydantic_model__ = create_model(cls.__name__,
+                                          __config__=inherit_config(config, BaseConfig),
+                                          **fields)
+    cls.__pydantic_model__.__config__ = cls.__pydantic_model__.config
```

Comment:
I think this is only necessary due to an error in `create_model`. On line 427 of `main.py` it should b `'__config__': config,` not `'config': ...`.

Please could you fix it in `main.py`. I've just checked locally and one tests needs tweaking, but otherwise not changes are required.

Reply:
Ok